api/spec/models/spree/api/webhooks/stock_movement_decorator_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'spec_helper'
+
+describe Spree::StockMovement do
+  let(:stock_item) { create(:stock_item) }
+  let(:variant) { stock_item.variant }
+  let(:stock_movement) { create(:stock_movement, stock_item: stock_item, quantity: quantity) }
+  let(:body) { Spree::Api::V2::Platform::VariantSerializer.new(variant).serializable_hash.to_json }
+
+  describe '#update_stock_item_quantity' do
+    subject { stock_movement }
+
+    describe 'when the variant goes out of stock' do
+      let(:quantity) { -stock_item.count_on_hand }
+
+      it 'emits the variant.out_of_stock event' do
+        expect { subject }.to emit_webhook_event('variant.out_of_stock')
+      end
```

Comment:
@aleksandarpetrushev let's also add a test when count_on_hand is 0 and there is an update which still persists it to 0